web-platform/website/src/components/pages/studies/study-card/ToastComponent.tsx

```diff
@@ -65,10 +68,20 @@ export function ToastComponent({
           <div className="right d-flex align-items-center">
             <Button
               onClick={async () => {
-                openModal ? await sendEmailVerification() : null;
-                openModal && (openModal as () => void)();
-                openPrivacyModal && (openPrivacyModal as () => void)();
-                setToastVisibility(false);
+                if (type === strings.verifyEmail.type) {
+                  try {
+                    await sendEmailVerification();
+                  } catch (e) {
+                    console.error(
+                      "Error when trying to resend email verification:",
+                      e
+                    );
+                  }
+                  (onOpenModal as () => void)();
```

Comment:
This is where we are breaking generic nature of this component. ToastComponent is supposed to represent a generic toast and adding email verification does not make sense here. This should be done in onOpenModal from ProductToasts IMO.